### ISSUE TITLE
Implement outcome regression as test

### ISSUE THREAD

**Issue by alejandroschuler** [rieszreg/rieszreg#36]
In Ollie's paper it reads

To show the generality of the Riesz representer setup, consider Setting \ref{setting:alf}, with $m^{\text{(OR)}}(f, x, y) = f(x)y$. In this setting the Riesz representer is the outcome regression function $\alpha_0^{\text{(OR)}} = \mu_0$ (with $\mu_0(x)=E_{P_0}[Y  \mid X=x]$ as above), since for all $f \in \mathcal{H}$
$$E_{P_0}[m^{\text{(OR)}}(f, X, Y)] = E_{P_0}[f(X)Y] = \langle f, \alpha_0^{\text{(OR)}} \rangle.$$
Thus, one can view outcome regression as learning the Riesz representer for the bounded linear functional $f \mapsto E[f(X) Y]$. In some sense, therefore, Riesz representer learning generalizes outcome regression to a broader class of statistical functionals.

Under Riesz loss, we get

$$P_n[ \alpha^2 - 2m(\alpha) ]= \P_n [ \alpha^2 - y \alpha ] $$

which in argmin is equal to 

$$\P_n [ \alpha^2 - y \alpha +y^2] = \P_n [ (\alpha - y)^2],$$

simply MSE. Furthermore, the data-augmentation recipe applied to this estimand yields the original dataset back (verify this formally). Therefore minimizing Riesz loss for this estimand should be exactly equivalent to outcome regression.

This is therefore a very useful test: we should implement this estimand and test our methods with it on all our different learner backends against the standard matching sklearn/xgboost/torch regression algorithm and make sure the results are essentially the same for the same hyperparameters.
